lumicks/pylake/tests/test_imaging_camera/test_image_stack.py

```diff
@@ -121,6 +122,25 @@ def test_image_stack(shape):
     np.testing.assert_allclose([x.start for x in stack[2]], [30])
 
 
+def test_stack_from_dataset():
+    frame_times = make_frame_times(2)
+    mock_stack = TiffStack(
+        [MockTiffFile([np.zeros((3, 3, 3))] * 2, frame_times)],
+        align_requested=False,
+    )
+
+    stack = ImageStack.from_dataset(mock_stack, name="hello", start_idx=1, stop_idx=2, step=3)
+    assert stack.name == "hello"
+    assert stack._start_idx == 1
+    assert stack._stop_idx == 2
+    assert stack._step == 3
+
+
+def test_stack_name(rgb_tiff_file):
+    cs = ImageStack(str(rgb_tiff_file), align=True)
+    cs._src.close()
```

Comment:
why do you need to close the source? or alternatively, why don't you need to in the previous test?

Reply:
Otherwise the file stays open. This test loads a file on disk and the previous doesn't (it uses a mock).